#hcontortus_genome

This repository contains data and the analysis workflow for the manuscript "Extensive genomic and transcriptomic variation defines
the chromosome-scale assembly of *Haemonchus contortus*, a model gastrointestinal worm" (in review).

Preprint available here: https://www.biorxiv.org/content/10.1101/616672v3

Final paper available here: (when it is published...)

Code for the analysis is here:

## Abstract
### Background
*Haemonchus contortus* is a globally distributed and economically important gastrointestinal pathogen of small ruminants and has become the key nematode model for studying anthelmintic resistance and other parasite-specific traits among a wider group of parasites including major human pathogens. Two draft genome assemblies for *H. contortus* were reported in 2013; however, both were highly fragmented, incomplete, and differed from one another in important respects. While the introduction of long-read sequencing has significantly increased the rate of production and contiguity of de novo genome assemblies broadly, achieving high-quality genome assemblies for small, genetically diverse, outcrossing eukaryotic organisms such as *H. contortus* remains a significant challenge.

### Results
Here, we report using PacBio long read and OpGen and 10X Genomics long-molecule methods to generate a highly contiguous 283.4 Mbp chromosome-scale genome assembly including a resolved sex chromosome. We show a remarkable pattern of almost complete conservation of chromosome content with *Caenorhabditis elegans*, but almost no conservation of gene order. Short and long-read transcriptome sequence data has allowed us to define coordinated transcriptional regulation throughout the life cycle of the parasite, and refine our understanding of *cis-* and *trans-* splicing relative to that observed in *C. elegans*. Finally, we use this assembly to give a comprehensive picture of chromosome-wide genetic diversity both within a single isolate and globally.

### Conclusions
The *H. contortus* MHco3(ISE).N1 genome assembly presented here represents one of the most contiguous and resolved nematode assemblies outside of the Caenorhabditis genus to date, together with one of the highest-quality set of predicted gene features. These data provide a high-quality comparison for understanding the evolution and genomics of Caenorhabditis and other nematodes and extends the experimental tractability of this model parasitic nematode in understanding pathogen biology, drug discovery and vaccine development, as well as important adaptive traits such as drug resistance.
